xc7/fasm2bels/iob_models.py

```diff
@@ -94,53 +94,52 @@ def append_obuf_iostandard_params(
 ):
     """
     Appends IOSTANDARD, DRIVE and SLEW parameters to the bel. The IOSTANDARD
-    and DRIVE parameters have to be explicitly provided in the
-    top.iostandard_defs dict. If parameters from the dict contradicts those
-    decoded from fasm, an error is printed.
+    and DRIVE parameters have to be read from an EBLIF file. If parameters
+    from the EBLIF contradicts those decoded from fasm, an error is printed.
     """
 
-    # Check if we have external information about the IOSTANDARD and DRIVE of
-    # that site
-    if site.site.name in top.iostandard_defs:
-        iostd_def = top.iostandard_defs[site.site.name]
+    # Check if we have IO settings information for the site read from EBLIF
+    iosettings = top.get_site_iosettings(site.site.name)
+
+    # We don't. Use the default IOSTANDARD
+    if iosettings is None:
+        iosettings = {"IOSTANDARD": top.default_iostandard}
+
+    iostandard = iosettings.get("IOSTANDARD", None)
+    drive = iosettings.get("DRIVE", None)
+
+    # Check if this is possible according to decoded fasm
+    is_valid = (iostandard, drive, slew) in possible_iostandards
+    if not is_valid:
+        eprint(
+            "IOSTANDARD+DRIVE+SLEW settings provided for {} do not match "
+            "their counterparts decoded from the fasm".format(site.site.name)
+        )
 
-        iostandard = iostd_def["IOSTANDARD"]
-        drive = iostd_def["DRIVE"] if "DRIVE" in iostd_def else None
+        eprint("Requested:")
+        eprint(" IOSTANDARD={}, DRIVE={}".format(iostandard, drive))
 
-        # Check if this is possible according to decoded fasm
-        is_valid = (iostandard, drive, slew) in possible_iostandards
-        if not is_valid:
+        eprint("Candidates are:")
+        eprint(" IOSTANDARD        | DRIVE  | SLEW |")
+        eprint("-------------------|--------|------|")
+        for i, d, s in possible_iostandards:
             eprint(
-                "IOSTANDARD+DRIVE+SLEW settings provided for {} do not match "
-                "their counterparts decoded from the fasm".format(
-                    site.site.name
+                " {}| {}| {}|".format(
+                    i.ljust(18),
+                    str(d).ljust(7), s.ljust(5)
                 )
             )
+        eprint("")
 
-            eprint("Requested:")
-            eprint(" IOSTANDARD={}, DRIVE={}".format(iostandard, drive))
-
-            eprint("Candidates are:")
-            eprint(" IOSTANDARD        | DRIVE  | SLEW |")
-            eprint("-------------------|--------|------|")
-            for i, d, s in possible_iostandards:
-                eprint(
-                    " {}| {}| {}|".format(
-                        i.ljust(18),
-                        str(d).ljust(7), s.ljust(5)
-                    )
-                )
-            eprint("")
+        # Demote NSTD-1 to warning
+        top.disable_drc("NSTD-1")
```

Comment:
Under what circumstances should the code ever reach this path?  Why is this not a failure?

Reply:
Still wondering about this?